CMakeLists.txt

```diff
@@ -1293,7 +1293,8 @@ if(MSVC)
   if(WARNINGS_PEDANTIC)
     target_compile_options(mixxx-lib PUBLIC /W4)
   else()
-    target_compile_options(mixxx-lib PUBLIC /W3)
+    # Warning Level 3 (production quality) / Treat Warnings as Errors
+    target_compile_options(mixxx-lib PUBLIC /W3 /WX)
```

Comment:
We already have the /WX flag, enabled by the WARNINGS_FATAL switch. The reason is that this allows to try out things without being forced to fix all warnings during development. The Flag is set in case of Ubuntu in the CI that prevents that any warning slips into the code. This is very convenient. 
So instead adding the fag here, please enable WARNINGS_FATAL in .github/workflow/build.yml

Reply:
But than you don't notice it in local builds. It's very inconvinient if the local build passes and CI fails.

Reply:
Remember that warnings can also happen after a compiler update or if you use a different compiler than the CI.  
My IDE shows me the warnings inline in the code. It is hard to miss that. But you can anyway configure WARNINGS_FATAL as you like. I thing we should stick with that approach for all platforms.